### PR TITLE
feat: Likes feat, Bookmarks & Profile pages in Social Media

### PR DESCRIPTION
**Pull request by kedark152**
### feat: Likes feat, Bookmarks & Profile pages in Social Media.

Pages to check: Home Page, Bookmarks Page & Profile page.
Features added:-

- Added Likes Feature to Home, Bookmark & Profile pages.
- Added Post Bookmark feature.
- Added Separate page to show bookmarks posts.
- Updated Who to Follow Section.
- Added Edit Profile Details for changing Full name, Profile Image, Cover Image, website & bio details.
- Learnt Firebase Storage feature to upload images  to get dynamic links.
- added logout feature to profile page.
- Updated Cover Image with Profile.

Preview Link: https://deploy-preview-4--agrolife.netlify.app/

### CODE REVIEW

File: src/utilities/statusesConstants.js [hunk 1]
```diff
@@ -8,4 +8,14 @@ export const STATUSES = Object.freeze({
...
   POST_DELETED: 'Post Deleted',
   POST_EDITED: 'Post Edited',
+  IS_FOLLOWING: 'Is Following',
+  IS_UNFOLLOWING: 'Is Unfollowing',
+  SUCCESS_LIKE: 'success_like',
+  SUCCESS_UNLIKE: 'success_unlike',
+  SUCCESS_BOOKMARK: 'success_bookmark',
+  SUCCESS_UNBOOKMARK: 'success_unbookmark',
+  POST_LIKED: 'Post Liked',
+  POST_UNLIKED: 'Post Unliked',
```
Comment (devananth) [inline src/utilities/statusesConstants.js:18]: - Good way of storing the constants

File: src/pages/MyProfile.jsx [hunk 2]
```diff
@@ -17,18 +19,96 @@ import {
...
+      dispatch(setUnFollowingState(userId));
+    } else {
+      dispatch(followUser({ currentUserId, followUserId: userId }));
+      dispatch(setFollowingState(userId));
+    }
+  };
+
+  useEffect(() => {
+    dispatch(getUserProfile({ userId }));
+    // dispatch(setUserProfileStatus('idle'));
```
Comment (devananth) [inline src/pages/MyProfile.jsx]: comment lines can be removed

File: src/features/users/userSlice.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+      console.log('getUserProfile Rejected', action.payload);
+      state.userProfileStatus = STATUSES.ERROR;
+    },
+    [followUser.pending]: state => {
+      state.followStatus = STATUSES.LOADING;
+    },
+    [followUser.fulfilled]: state => {
+      state.followStatus = STATUSES.SUCCESS;
+      state.followStatusMessage = STATUSES.IS_FOLLOWING;
+      // state.allUser = action.payload;
```
Comment (devananth) [inline src/features/users/userSlice.js]: this can be removed

**Review by devananth — APPROVED** [review]
- Overall code looks well structured and clean
- Try removing the commented codes and consoles

**Review by bajajcodes — COMMENTED** [review]
- Code is consistent and readable.
- Removed comments or specify reason for comments
- Good use of hoooks

File: src/components/EditProfileModal.jsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    headerImgURL:
+      coverImageUrl.length == 0
+        ? 'https://tiny.cc/defaultHeaderImg'
+        : coverImageUrl,
+    profileImgURL: profileImageUrl,
+    headerImg: null,
+    profileImg: null,
+  });
+
+  const handleMediaPreview = e => {
```
Comment (ankit9015) [inline src/components/EditProfileModal.jsx:65]: I like that you are limiting the image file size

File: src/components/EditProfileModal.jsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+      <Modal initialFocusRef={initialRef} isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Edit Profile Details</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <Box position="relative">
+              <Image
+                // src={headerImage}
```
Comment (ankit9015) [inline src/components/EditProfileModal.jsx:102]: Remove commented code if not needed

File: src/components/cards/WhoToFollow.jsx [hunk 6]
```diff
@@ -1,72 +1,83 @@
...
+import { useDispatch } from 'react-redux';
 
 export const WhoToFollow = () => {
+  const dispatch = useDispatch();
+  const authState = useSelector(state => state.auth);
+  const userId = authState.userData.uid;
+  const followingList = authState.userData.following;
+  const allUser = useSelector(state => state.user.allUser);
+  const showAllUsers = allUser.filter(user => user.uid !== userId);
+  //&& !followingList.includes(user.uid)
```
Comment (ankit9015) [inline src/components/cards/WhoToFollow.jsx:26]: Remove commented code

File: src/features/users/userSlice.js [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+  }
+);
+
+const initialState = {
+  allUser: [],
+  allUserStatus: STATUSES.IDLE,
+  followStatus: STATUSES.IDLE,
+  followStatusMessage: '',
+  userProfile: {
+    // uid,
```
Comment (ankit9015) [inline src/features/users/userSlice.js:154]: remove commented code

**Review by ankit9015 — APPROVED** [review]
Overall the code is clean and ready to merge.

**Review by ankit9015 — APPROVED** [review]
Overall the code is clean and ready to merge.

**Comment by kedark152** [comment]
Thanks all for PR Review, removed commented code and made changes.

**Pull request merged**